cpp/src/arrow/dataset/test_util.h

```diff
@@ -823,6 +850,34 @@ class FileFormatScanMixin : public FileFormatFixtureMixin<FormatHelper>,
       ASSERT_EQ(row_count, expected_rows());
     }
   }
+
+  void TestScanWithFieldPathFilter() {
+    compute::Expression index_filter = equal(field_ref(0), literal(0));
+    compute::Expression name_filter = equal(field_ref("i64"), literal(0));
+    std::vector<int64_t> row_counts(2);
```

Comment:
This is wrong when used with push_back, the end result will be a vector `[0, 0, x, y]`